hoomd/ForceCompute.cc

```diff
@@ -336,26 +343,39 @@ pybind11::object ForceCompute::getEnergiesPython()
         {
         dims[0] = 0;
         }
-    std::vector<double> energy(dims[0]);
+    std::vector<double> global_energy(dims[0]);
 
-    // This is slow: TODO implement a proper gather operation
-    for (unsigned int i = 0; i < m_pdata->getNGlobal(); i++)
+    // sort energies by particle tag
+    std::vector<double> local_energy;
+    local_energy.reserve(m_pdata->getN());
+    std::vector<uint32_t> local_tag(m_pdata->getN());
+    ArrayHandle<unsigned int> h_tag(m_pdata->getTags(), access_location::host, access_mode::read);
+    ArrayHandle<unsigned int> h_rtag(m_pdata->getRTags(), access_location::host, access_mode::read);
+    ArrayHandle<Scalar4> h_force(m_force, access_location::host, access_mode::read);
+    std::copy(h_tag.data, h_tag.data + m_pdata->getN(), local_tag.begin());
+    std::sort(local_tag.begin(), local_tag.end());
+    for (unsigned int i = 0; i < m_pdata->getN(); i++)
         {
-        double e = getEnergy(i);
-        if (root)
-            {
-            energy[i] = e;
-            }
+        local_energy.push_back(h_force.data[h_rtag.data[local_tag[i]]].w);
         }
 
-    if (root)
+    if (m_sysdef->isDomainDecomposed())
         {
-        return pybind11::array(dims, energy.data());
+#ifdef ENABLE_MPI
+        m_gather_tag_order.setLocalTagsSorted(local_tag);
+        m_gather_tag_order.gatherArray(global_energy, local_energy);
+#endif
         }
     else
         {
-        return pybind11::none();
+        global_energy = local_energy;
```

Comment:
```suggestion
        global_energy = std::move(local_energy);
```
I think this will perform a copy? Should we do an explicit `std::move`?